lib/absinthe/subscription/phase_serializer.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Absinthe.Subscription.PipelineSerializer do
+  @moduledoc """
+  Serializer responsible for packing and unpacking pipeline stored in the Elixir registry.
+
+  The purpose of this logic is saving memory by deduplicating repeating options - (ETS
+  backed registry stores them flat in the memory).
+  """
+
+  alias Absinthe.{Phase, Pipeline}
+
+  @type options_label :: {:options, non_neg_integer()}
+
+  @type packed_phase_config :: Phase.t() | {Phase.t(), options_label()}
+
+  @type options_map :: %{options_label() => Keyword.t()}
+
+  @type packed_pipeline :: {:packed, [packed_phase_config()], options_map()}
+
+  @spec pack(Pipeline.t()) :: packed_pipeline()
+  def pack(pipeline) do
+    {reverse_pipeline, options_reverse_map} =
+      pipeline
+      |> List.flatten()
+      |> Enum.reduce({[], %{}}, fn phase, {pipeline, options_reverse_map} ->
```

Comment:
Might as well use `map_reduce` here to avoid the reverse at the end.

Reply:
Addressed in https://github.com/absinthe-graphql/absinthe/pull/1006/commits/67acc31e133ee46349a84f24ef4b4680f7d17322